java/ql/src/experimental/Security/CWE/CWE-352/JsonpInjectionLib.qll

```diff
@@ -0,0 +1,115 @@
+import java
+import DataFlow
+import JsonStringLib
+import semmle.code.java.security.XSS
+import semmle.code.java.dataflow.DataFlow
+import semmle.code.java.dataflow.DataFlow3
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.frameworks.spring.SpringController
+
+/**
+ * A method that is called to handle an HTTP GET request.
+ */
+abstract class RequestGetMethod extends Method {
+  RequestGetMethod() {
+    not exists(MethodAccess ma |
+      // Exclude apparent GET handlers that read a request entity, because this is the principle of JSONP.
```

Comment:
```suggestion
      // Exclude apparent GET handlers that read a request entity, because this likely indicates this is not in fact a GET handler.
      // This is particularly a problem with Spring handlers, which can sometimes neglect to specify a request method.
      // Even if it is in fact a GET handler, such a request method will be unusable in the context `<script src="...">`,
      // which is the typical use-case for JSONP but cannot supply a request body.
```